### ISSUE TITLE
Can't delete imported csv files.

### ISSUE THREAD

**Issue by chreichwein** [snipe/snipe-it#5034]
Let's make this easy.

See my screenshots:

![3](https://user-images.githubusercontent.com/36449487/36308550-762a0c80-1321-11e8-831a-285110a7f909.JPG)

![2](https://user-images.githubusercontent.com/36449487/36308554-7b113a48-1321-11e8-84c0-cee09b381580.JPG)

![1](https://user-images.githubusercontent.com/36449487/36308558-7efaaf9a-1321-11e8-96fa-562cadb59b49.JPG)

Where can I change: 
data-url="http://xxx/api/v1/imports"

To: 
data-url="http://xxx/inetpub/wwwroot/snipe-it/storage/private_uploads/imports"

Or is this an other issue?

PS: I really love your Snipe IT assetmanager <3 Thanks for your awesome work and love you put in!

- [x ] I have enabled debug mode 
- [x ] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

Fresh install
Snipe IT Version v4.1.11-pre - build 3307
PHP 7.1
MariaDB 10.0
Windows Server 2012 R2 IIS
Manual installation

**Comment by snipe** [comment]
Duplicate of #4718

**Comment by snipe** [comment]
Sorry, was flying on autopilot there. 

> Where can I change: data-url="http://xxx/api/v1/imports"
> To: data-url="http://xxx/inetpub/wwwroot/snipe-it/storage/private_uploads/imports"

I'm not sure why you would want to change that. 

Does that CSV file actually exist in your filesystem?

**Comment by chreichwein** [comment]
Yep, all csv files exist in the imports dir. I just thought the path is wrong?  This is the only error I can see (chrome console). It can't delete the file, I also have all the needed rights set to the folders.

**Comment by snipe** [comment]
The url is correct  - it points to the script that handles the file deletion.

@dmeltzer any idea why they're getting method not allowed? I can't reproduce this locally.

**Comment by dmeltzer** [comment]
Does an entry with the id of 18 exist in the imports table?

**Comment by chreichwein** [comment]
Well, let me check this on Monday.

**Comment by snipe** [comment]
@dmeltzer that still seems like a weird error though.

**Comment by chreichwein** [comment]
Hi there,

yes all ID's "18" exists: 
![image](https://user-images.githubusercontent.com/36449487/36479877-9958f328-170a-11e8-9949-a5d9e53565e6.png)

**Comment by chreichwein** [comment]
Chrome Developer Tools show me this, when I click the delete button:

(debugger)
(function() {
with(this){return _c('div',{attrs:{"id":"app"}},[_c('importer',{inlineTemplate:{render:function(){with(this){return _c('div',{staticClass:"row"},[_c('alert',{directives:[{name:"show",rawName:"v-show",value:(alert.visible),expression:"alert.visible"}],attrs:{"alert-type":alert.type},on:{"hide":function($event){alert.visible = false}}},[_v(_s(alert.message))]),_v(" "),_c('errors',{attrs:{"errors":importErrors}}),_v(" "),_c('div',{staticClass:"col-md-12"},[_c('div',{staticClass:"box"},[_c('div',{staticClass:"box-body"},[_c('div',{staticClass:"row"},[_m(0),_v(" "),_c('div',{directives:[{name:"show",rawName:"v-show",value:(progress.visible),expression:"progress.visible"}],staticClass:"col-md-9",staticStyle:{"padding-bottom":"20px"}},[_c('div',{staticClass:"col-md-11"},[_c('div',{staticClass:"progress progress-striped-active",staticStyle:{"margin-top":"8px"}},[_c('div',{staticClass:"progress-bar",class:progress.currentClass,style:(progressWidth),attrs:{"role":"progressbar"}},[_c('span',[_v(_s(progress.statusText))])])])])])]),_v(" "),_c('div',{staticClass:"row"},[_c('div',{staticClass:"col-md-12",staticStyle:{"padding-top":"30px"}},[_c('table',{staticClass:"table table-striped",attrs:{"id":"upload-table"}},[_m(1),_v(" "),_c('tbody',[_l((files),function(currentFile){return [_c('tr',[_c('td',[_v(_s(currentFile.file_path))]),_v(" "),_c('td',[_v(_s(currentFile.created_at)+" ")]),_v(" "),_c('td',[_v(_s(currentFile.filesize))]),_v(" "),_c('td',[_c('button',{staticClass:"btn btn-sm btn-info",on:{"click":function($event){toggleEvent(currentFile.id)}}},[_v("Process")]),_v(" "),_c('button',{staticClass:"btn btn-sm btn-danger",on:{"click":function($event){deleteFile(currentFile)}}},[_c('i',{staticClass:"fa fa-trash icon-white"})])])]),_v(" "),_c('import-file',{key:currentFile.id,attrs:{"file":currentFile,"custom-fields":customFields},on:{"alert":function($event){updateAlert(alert)}}})]})],2)])])])])])])],1)}},staticRenderFns:[function(){with(this){return _c('div',{staticClass:"col-md-3"},[_c('span',{staticClass:"btn btn-info fileinput-button"},[_c('span',[_v("Select Import File...")]),_v(" "),_c('input',{attrs:{"id":"fileupload","type":"file","name":"files[]","data-url":"http://ger1srv08/api/v1/imports","accept":"text/csv"}})])])}},function(){with(this){return _c('thead',[_c('tr',[_c('th',[_v("File")]),_v(" "),_c('th',[_v("Created")]),_v(" "),_c('th',[_v("Size")]),_v(" "),_c('th')])])}}]}})],1)}
})

Maybe this is helpful.

![image](https://user-images.githubusercontent.com/36449487/36480185-a9f8fa74-170b-11e8-8477-94d55be855e0.png)

**Comment by cepacs** [comment]
I am noticing the same behavior too.. blank orange bar with Method not allowed
![image](https://user-images.githubusercontent.com/36515590/36489765-c8c76ed4-16f4-11e8-9974-d36b85f25353.png)

I do see it in the imports table
![image](https://user-images.githubusercontent.com/36515590/36489844-f892c24e-16f4-11e8-98a7-eab38e9850ae.png)

And the file is in the correct folder
![image](https://user-images.githubusercontent.com/36515590/36493514-cc403be6-16fd-11e8-8f1e-c88ca4227f17.png)

Fresh
Version v4.1.11-pre - build 3299 (master)
PHP version: 7.1.14
MySQL Server version: 5.7.21
Windows Server 2016 / IIS
GIT Install

**Comment by snipe** [comment]
I have been utterly unable to reproduce this locally or on any of the demos. :(

**Comment by cepacs** [comment]
Could this have to do with running it on a Windows OS maybe?

**Comment by snipe** [comment]
I wouldn't think so. 

```
/**
 * Remove the specified resource from storage.
 *
 * @param  \App\Import  $import
 * @return \Illuminate\Http\Response
 */
public function destroy($import_id)
{
    $this->authorize('create', Asset::class);
    $import = Import::find($import_id);
    try {
        unlink(config('app.private_uploads').'/imports/'.$import->file_path);
        $import->delete();
        return response()->json(Helper::formatStandardApiResponse('success', null, trans('admin/hardware/message.import.file_delete_success')));
    } catch (\Exception $e) {
        return response()->json(Helper::formatStandardApiResponse('error', null, trans('admin/hardware/message.import.file_delete_error')), 500);
    }
}
```

It's just using a standard PHP `unlink()`, nothing fancy. And the error is (weirdly) a method not allowed error, instead of a "couldn't delete file - not found" error.

**Comment by snipe** [comment]
@dmeltzer was it a deliberate thing to remove the file extensions from the CSVs once uploaded?

<img width="746" alt="screen shot 2018-02-21 at 1 22 52 pm" src="https://user-images.githubusercontent.com/197404/36506083-60ea6616-170a-11e8-9664-fc45b464c60b.png">

**Comment by dmeltzer** [comment]
No, I saw that a few weeks ago and never tracked it down..

**Comment by dmeltzer** [comment]
Ah, thats actually a result of 3eea1a23f8, sluggifying the filename.  the slug removes the dot in the extension name.

Not sure if that srelated.. though on windows it could be?

**Comment by colinerd** [comment]
I have resolved this issue and would like to share it with you:

It seems to be an issue with IIS and thus on Windows hosting. You will have to go into IIS manager, select your site (or do it on the server to apply to all sites) and open "Handler Mappings" - From here, select your PHP mappings (there could be 2) and open it. Click op "Request Restrictions" and select the tab "Verbs".

Here you will have to add DELETE. Mine looks like this now "GET,HEAD,POST,DELETE"
Save the changes (it might complain you need to use quotes on the .exe file, something I ran into)
Do this for your other php entry under "Handler Mappings" as well and when done, restart IIS server.

This change will allow the DELETE method to be used and was the reason why you will get a "method not allowed". This resolved the issue for me and I was able to delete all my import files after using them.

**Comment by blackrandom** [comment]
@colinerd Thank you! I followed your directions and I'm finally able to delete the imports now! And yeah this does seem like it should be added as part of the installation procedure for IIS as it doesn't give any useful errors on the site when you try to delete.

**Comment by cepacs** [comment]
Same here.. worked great.  Thanks for figuring that out for us IIS people. :+1: 

The only odd thing was if I deleted one file the next one said "Unable to delete file", but if I refreshed the page I can delete another file.  So I just refreshed between deletes and it all was good.. :)

**Comment by dmeltzer** [comment]
Thanks for tracking this down and figuring it out.  I would not have guessed that IIS may not support those HTTP verbs.

For future proofing you may want to enable PUT and PATCH as well.  The snipe-it api uses them currently, and the frontend may begin using them more regularally as development continues.

**Comment by colinerd** [comment]
Thanks, I’ll also add those 2 to my verbs list to future proof my installation.

The instructions look good to me :-)

**Comment by chreichwein** [comment]
Thanks, this works perfectly now!

**Comment by cepacs** [comment]
@dmeltzer The instructions look good. Thanks for updating it :+1:

**Comment by cepacs** [comment]
The only thing I will add is when I modify the verbs here I get an Edit Module Mapping message about the handler must be a .dll or .exe when I click OK.  I had to just add double quotes around the Executable field.

So instead of C:\Program Files (x86)\PHP\v5.3\php-cgi.exe I did "C:\Program Files (x86)\PHP\v5.3\php-cgi.exe"

Then I could click OK and it asks If I'd liek to create a FastCGI application for this executable and I choose Yes.

Not sure if this only happens to me but it may help somebody else. :)

**Comment by blackrandom** [comment]
@cepacs I can confirm, the same issue happened for me as well. Putting the double quotes resolves the issue.

**Comment by snipe** [comment]
@colinerd Is it possible for you to provide some screenshots so we can update the docs with this excellent info?

**Comment by cepacs** [comment]
I was on our server and did some screen shots.  These are in the same order as I got them, along with the error when the double quotes are not there..  Obviously this is only for one of the PHP handlers.  We are on IIS 10 under Windows Server Standard 2016.

![image](https://user-images.githubusercontent.com/36515590/37789697-00d6eadc-2ddb-11e8-9893-8a84f5288f31.png)

![image](https://user-images.githubusercontent.com/36515590/37789713-06a5aec6-2ddb-11e8-8e9e-4cbfd6ba3f5f.png)

![image](https://user-images.githubusercontent.com/36515590/37789726-0ba28732-2ddb-11e8-8c6c-2779c8448b8a.png)

![image](https://user-images.githubusercontent.com/36515590/37789736-0ff718e8-2ddb-11e8-8782-679564a00e3f.png)

![image](https://user-images.githubusercontent.com/36515590/37789739-146af016-2ddb-11e8-9953-6f3a6856a6f6.png)

![image](https://user-images.githubusercontent.com/36515590/37789749-1812e19c-2ddb-11e8-8e94-a92a167390c3.png)

![image](https://user-images.githubusercontent.com/36515590/37789761-1c49d6c6-2ddb-11e8-89f3-6755a9852fe6.png)

![image](https://user-images.githubusercontent.com/36515590/37789773-20a8f49a-2ddb-11e8-9ad0-13e2ba3c381d.png)

**Comment by snipe** [comment]
@cepacs This is PERFECT. Thank you SO much! I've updated the docs: https://snipe-it.readme.io/v4.1.14/docs/windowsiis#section-set-additional-handler-mappings

**Comment by cepacs** [comment]
Glad it worked.. I noticed that the image under "Select a PHP mapping (there may be more than one) and open it." didn't transfer properly to the doc..

**Comment by eemjema** [comment]
I encountered a similar issue although am developing on WSL2. After some debugging, I realized that in my case, the problem was caused by setting `PRIVATE_FILESYSTEM_DISK=s3_private ` and `PUBLIC_FILESYSTEM_DISK=s3_public`. The imports were still being uploaded locally, but when attempting to delete them, the system tried to delete them from S3. I resolved this by specifying the disk name explicitly. I changed: `if (Storage::delete('private_uploads/imports/'.$file->file_path))` to `if (Storage::disk('local')->delete('private_uploads/imports/'.$file->file_path))` in the importer.
